### PR TITLE
core/bitarthm: suppress false positives

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

As the title says.

### Testing procedure

The CI will run `static-tests`

### Issues/PRs references

None

### CODE REVIEW

**Comment by maribu** [comment]
Doesn't change code and static tests are also run without Murdock

**Comment by chrysn** [comment]
Checked with [the error definition](    reportError(errors, Severity::style, "oppositeExpression", "Opposite expression on both sides of \'" + op + "\'.\n") -- indeed this does not apply here (or, more precisely: we *did* check the code carefully).

**Review by chrysn — APPROVED** [review]

File: core/include/bitarithm.h [hunk 1]
```diff
@@ -172,6 +172,7 @@ static inline unsigned bitarithm_lsb(unsigned v)
 {
 /* Source: http://graphics.stanford.edu/~seander/bithacks.html#ZerosOnRightMultLookup */
     extern const uint8_t MultiplyDeBruijnBitPosition[32];
+    /* cppcheck-suppress oppositeExpression */
```
Comment (miri64) [inline core/include/bitarithm.h:175]: Reasoning is missing.

File: core/include/bitarithm.h [hunk 2]
```diff
@@ -216,6 +217,7 @@ static inline unsigned bitarithm_test_and_clear(unsigned state, uint8_t *index)
 #elif defined(BITARITHM_LSB_LOOKUP)
     /* Source: http://graphics.stanford.edu/~seander/bithacks.html#ZerosOnRightMultLookup */
     extern const uint8_t MultiplyDeBruijnBitPosition[32];
+    /* cppcheck-suppress oppositeExpression */
```
Comment (miri64) [inline core/include/bitarithm.h:220]: Here too.

**Review by miri64 — CHANGES_REQUESTED** [review]

File: core/include/bitarithm.h (hunk 1 above)
Comment (chrysn) [inline core/include/bitarithm.h:175]:
```suggestion
    /* cppcheck-suppress oppositeExpression
     * (reason: `x & -x` is a bit twiddling idiom to extract the LSB; the
     * check assumes that all arguments to binary operations should have the
     * same sign) */
```

File: core/include/bitarithm.h (hunk 2 above)
Comment (chrysn) [inline core/include/bitarithm.h:220]:
```suggestion
    /* cppcheck-suppress oppositeExpression
     * (reason: `x & -x` is a bit twiddling idiom to extract the LSB; the
     * check assumes that all arguments to binary operations should have the
     * same sign) */
```

**Comment by chrysn** [comment]
My ACK holds and is now better founded with the comments; @miri64, you happy?

File: core/include/bitarithm.h [hunk 3]
```diff
@@ -172,6 +172,10 @@ static inline unsigned bitarithm_lsb(unsigned v)
 {
 /* Source: http://graphics.stanford.edu/~seander/bithacks.html#ZerosOnRightMultLookup */
     extern const uint8_t MultiplyDeBruijnBitPosition[32];
+    /* cppcheck-suppress oppositeExpression
+     * (reason: `x & -x` is a bit twiddling idiom to extract the LSB; the
+     * check assumes that all arguments to binary operations should have the
+     * same sign) */
```
Comment (maribu) [inline core/include/bitarithm.h]: Actually, this is not about the same sign. I changed the wording a bit, please confirm.

Comment (chrysn) [reply core/include/bitarithm.h]: Yeah, this is better, the old wording was sloppy.

**Review by miri64 — APPROVED** [review]

**Pull request merged**